modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseUpgradeCheck.java

```diff
@@ -46,13 +54,85 @@ protected String addNewImports(String newContent) {
 		return newContent;
 	}
 
+	protected String addNewImportsJSP(String newContent, String... newImports) {
+		if (newImports.length == 0) {
+			return newContent;
+		}
+
+		Set<String> missingImports = new TreeSet<>();
+
+		Collections.addAll(missingImports, newImports);
+
+		if (missingImports.isEmpty()) {
+			return newContent;
+		}
+
+		Matcher directiveMatcher = _directivePattern.matcher(newContent);
+
+		if (directiveMatcher.find()) {
+			String includeDirective = directiveMatcher.group();
+
+			String[] includeDirectives = StringUtil.splitLines(
+				includeDirective);
+
+			String newHeaderPackage = createImportsPackageJsp(
+				missingImports, includeDirectives);
+
+			newContent = StringUtil.replaceFirst(
+				newContent,
+				com.liferay.petra.string.StringUtil.merge(
```

Comment:
@TamyrisTorres, there is a reason to use the petra StringUtil?

Reply:
Yes, because I need to replace the String[] by String, and this method is just in Petra Stringutil.